proposal/color-4-new-spaces-js.d.ts.md

```diff
@@ -0,0 +1,864 @@
+
+```ts
+isChannelMissing(channel: ChannelName | 'alpha'): boolean;
+```
+
+[color.is-missing()]: ./color-4-new-spaces.md#coloris-missing-1
+
+#### `isAlphaMissing`
+
+Returns the result of [`color.is-missing(internal, 'alpha')`][color.is-missing()].
+
+```ts
+get isAlphaMissing(): boolean;
+```
+
+#### `isChannelPowerless`
+
+Returns the result of [`color.is-powerless(internal, channel, space)`].
+
+[`color.is-powerless(internal, channel, space)`]: ./color-4-new-spaces.md#coloris-powerless-1
+
+```ts
+isChannelPowerless(channel: ChannelName): boolean;
+isChannelPowerless(
+  channel: ChannelNameHSL,
+  options?: {space: ColorSpaceHSL}
+): boolean;
+isChannelPowerless(
+  channel: ChannelNameHWB,
+  options?: {space: ColorSpaceHWB}
+): boolean;
+isChannelPowerless(
+  channel: ChannelNameLAB,
+  options?: {space: ColorSpaceLAB}
+): boolean;
+isChannelPowerless(
+  channel: ChannelNameLCH,
+  options?: {space: ColorSpaceLCH}
+): boolean;
+isChannelPowerless(
+  channel: ChannelNameRGB,
+  options?: {space: ColorSpaceRGB}
+): boolean;
+isChannelPowerless(
+  channel: ChannelNameXYZ,
+  options?: {space: ColorSpaceXYZ}
+): boolean;
+```
+
+#### `interpolate`
+
+* Let `space` be the value of [`this.space()`].
+
+* If `options.method` is set, let `interpolationMethod` be a space separated
+  list containing the value of `space`, a space, and the value of `options.method`.
+
+* Otherwise, if `space` is a rectangular color space, let `interpolationMethod` be
+    `space`.
+  
+* Otherwise, let `interpolationMethod` be a space separated list containing th
+  value of `space`, a space, and the string "shorter".
+
+* Return the result of [`color.mix(internal, options.color2, options.weight, interpolationMethod)`][`color.mix()`].
+
+```ts
+interpolate(options: {color2: SassColor; weight?: number}): SassColor;
+
+interpolate(options: {
+  color2: SassColor;
+  weight?: number;
+  method?: HueInterpolationMethod;
+}): SassColor;
+```
+
+[`color.mix()`]: ./color-4-new-spaces.md#colormix-1
+
+### Updated Color Functions
+
+#### `change`
+
+Replace the definition of [color.change] with the following:
+
+[color.change]: ../spec/js-api/value/color.d.ts.md#change
+
+This algorithm takes a JavaScript object `options` and returns a new SassColor
+as the result of changing some of [`internal`]'s components.
+
+> The `space` value defaults to the `space` of [`internal`], and the caller may
+> specify any combination of channels and alpha in that space to be changed.
+>
+> If `space` is not a [legacy color space], a channel value of `null` will
+> result in a [missing component][missing components] value for that channel.
+
+* Let `initialSpace` be the value of [`this.space()`].
+
+* Let `spaceSetExplicitly` be `true` if `options.space` is defined, and `false`
+  otherwise.
+
+* Let `space` be `options.space` if `spaceSetExplicitly` is true, and the value of
+  `initialSpace` otherwise.
+
+* If `initialSpace` is a [legacy color space] and `options.space` is not set:
```

Comment:
Style nit: I'd either refer to `spaceSetExplicitly` here, or just refer directly to `options.space` being defined everywhere instead of using the intermediate variable.

Reply:
Resolved in 3599c5bdd932eef6b28a1e9a2a969656694fc510